### DISCUSSION TITLE
surround tag in visual mode

### DISCUSSION THREAD

**Discussion by timsofteng** [kylechui/nvim-surround#189] [General]
Hello. How to surround tag in visual mode like in vim-surround?

**Comment by kylechui** [comment]
You can use `S` or `gS` to add HTML-style tags in visual mode. The former adds tags exactly around the selected region, while the latter adds the tags on new lines. Additionally, the angle bracket boilerplate is handled automatically.

**Reply by brneor** [reply]
Not OP but I wanted the same thing. Tried your suggestion but I don't think I'm getting exactly what I want here... I'll try my best to explain:

What I want is
```
Old text                                Command         New text
<div>This is my |selection|</div>       ????            <div>This is my <span>selection</span></div>
```

What I can currently do:
```
Old text                    Command         New text
|div id="test"|</div>       S>              <div id="test"></div>
or
div cont*ents               ysstdiv         <div>div contents</div>
```

**Reply by kylechui** [reply]
Have you tried visually selecting the text, then typing `Stspan`?

**Reply by brneor** [reply]
Thanks, that did the trick :)

**Comment by 582727501** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我没在上网，看到后，我会尽快给您回复。

**Comment by 582727501** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我没在上网，看到后，我会尽快给您回复。
